### PR TITLE
PLU-40: Super minor edit to test step description

### PR DESCRIPTION
**Pull request by m0nggh**
Customised the error msg for formsg and this allows for more extensibility if raw webhook requires its own error msg!

## Before & After Screenshots

**BEFORE**:
image

**AFTER**:
image

### CODE REVIEW

**Review by ogp-weeloong — CHANGES_REQUESTED** [review]
This is not always for forms; we could have test data from other triggers such as webhook

File: packages/frontend/src/locales/en.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
   "flowEditor.unpublish": "UNPUBLISH",
   "flowEditor.publishedFlowCannotBeUpdated": "To edit this pipe, you must first unpublish it.",
   "flowEditor.noTestDataTitle": "We couldn't find matching data",
-  "flowEditor.noTestDataMessage": "Create a sample in the associated service and test the step again.",
+  "flowEditor.noTestDataMessage": "Make a new submission to the form you connected and test the step again.",
```
Comment (pregnantboy) [inline packages/frontend/src/locales/en.json]: can we revert this change? other than this lgtm

Comment (m0nggh) [reply packages/frontend/src/locales/en.json]: ok done, thanks!

**Review by pregnantboy — APPROVED** [review]
lgtm!

File: packages/frontend/src/components/TestSubstep/index.tsx [hunk 2]
```diff
@@ -142,7 +143,9 @@ function TestSubstep(props: TestSubstepProps): React.ReactElement {
               </AlertTitle>
 
               <Box sx={{ fontWeight: 400 }}>
-                {formatMessage('flowEditor.noTestDataMessage')}
+                {(selectedActionOrTrigger as IBaseTrigger)
```
Comment (ogp-weeloong) [inline packages/frontend/src/components/TestSubstep/index.tsx]: `hasNoOutput` can be true when `selectedActionOrTrigger` is an `IAction`, so we shouldn't cast to `IBaseTrigger` here.

File: packages/frontend/src/components/TestSubstep/index.tsx [hunk 3]
```diff
@@ -142,7 +143,9 @@ function TestSubstep(props: TestSubstepProps): React.ReactElement {
               </AlertTitle>
 
               <Box sx={{ fontWeight: 400 }}>
-                {formatMessage('flowEditor.noTestDataMessage')}
```
Comment (ogp-weeloong) [inline packages/frontend/src/components/TestSubstep/index.tsx:145]: This is the only place where `flowEditor.noTestDataMessage` is referenced; so if you remove this we don't need it in our localization `en.json` file anymore.

But don't remove it, I think we can re-use. I will suggest some changes.

File: packages/frontend/src/components/TestSubstep/index.tsx (hunk 2 above)
Comment (ogp-weeloong) [reply packages/frontend/src/components/TestSubstep/index.tsx]: If you were referencing the `(selectedActionOrTrigger as IBaseTrigger)` line above, that works because `step.webhookUrl` is only ever defined for triggers

_...although I don't think it's ideal, we need to shore up the typing in this codebase_

Comment (ogp-weeloong) [reply packages/frontend/src/components/TestSubstep/index.tsx]: That said... your code still works. But it's not ideal because its communicating the wrong types.

For fun, here's a convoluted one liner to do that in a type-safe manner using TS narrowing:

```
{selectedActionOrTrigger &&
  'webhookTriggerInstructions' in selectedActionOrTrigger &&
  selectedActionOrTrigger.webhookTriggerInstructions?.errorMsg
    ? selectedActionOrTrigger.webhookTriggerInstructions.errorMsg
    : formatMessage('flowEditor.noTestDataMessage')}
```

**Review by ogp-weeloong — APPROVED** [review]
LGTM but can we remove the cast to `IBaseTrigger` because it's not always a trigger.

Up to you if you want to keep `flowEditor.noTestDataMessage` or want to use your default trigger error as the "default error message".

File: packages/frontend/src/components/TestSubstep/index.tsx (hunk 2 above)
Comment (m0nggh) [reply packages/frontend/src/components/TestSubstep/index.tsx]: ok thanks a lot!
